### PR TITLE
[0.6/dx11] Fix Read-Only Depth Stencil

### PR DESCRIPTION
**Pull request by cwfitzgerald**
Depends on #3416. Commits will be rebased away once this is merged.

Fixes the water example under DX11.

### CODE REVIEW

File: src/backend/dx11/src/conv.rs [hunk 1]
```diff
@@ -730,6 +730,15 @@ pub(crate) fn map_depth_stencil_desc(
         None => unsafe { mem::zeroed() },
     };
 
+    let stencil_read_only = write_mask == 0 ||
```
Comment (kvark) [inline src/backend/dx11/src/conv.rs:733]: I wonder if we should make those conditions even more precise (at some point). I.e. technically speaking, if your stencil function is ALWAYS, then it doesn't matter what `StencilFailOp` is for any of the sides.

File: src/backend/dx11/src/device.rs [hunk 2]
```diff
@@ -633,6 +642,13 @@ impl Device {
         let mut desc: d3d11::D3D11_DEPTH_STENCIL_VIEW_DESC = unsafe { mem::zeroed() };
         desc.Format = info.format;
 
+        if let Some(stencil) = read_only_stencil {
+            desc.Flags = match stencil {
+                true => d3d11::D3D11_DSV_READ_ONLY_DEPTH | d3d11::D3D11_DSV_READ_ONLY_STENCIL,
```
Comment (kvark) [inline src/backend/dx11/src/device.rs:647]: there is never a case where we end up with just `d3d11::D3D11_DSV_READ_ONLY_STENCIL`? what if the format is stencil-only, and it's read-only?

**Review by kvark — COMMENTED** [review]
Once concern here. Great stuff overall!

File: src/backend/dx11/src/device.rs (hunk 2 above)
Comment (cwfitzgerald) [reply src/backend/dx11/src/device.rs:647]: DX11 doesn't seem to support stencil only formats, at least as far as I can tell.

File: src/backend/dx11/src/conv.rs (hunk 1 above)
Comment (cwfitzgerald) [reply src/backend/dx11/src/conv.rs:733]: Good point, shouldn't be too hard to add.

**Pull request merged**